src/lazy/encoder/value_writer.rs

```diff
@@ -204,13 +201,19 @@ pub(crate) use delegate_value_writer_to_self;
 pub struct FieldWriter<'field, StructWriterType> {
     name: RawSymbolRef<'field>,
     struct_writer: &'field mut StructWriterType,
+    value_writer_config: ValueWriterConfig,
 }
 
 impl<'field, StructWriterType> FieldWriter<'field, StructWriterType> {
-    pub fn new(name: RawSymbolRef<'field>, struct_writer: &'field mut StructWriterType) -> Self {
+    pub(crate) fn new(
+        name: RawSymbolRef<'field>,
+        value_writer_config: ValueWriterConfig,
+        struct_writer: &'field mut StructWriterType,
+    ) -> Self {
         Self {
             name,
             struct_writer,
+            value_writer_config,
```

Comment:
🗺️ Constructing a `FieldWriter` now takes a `ValueWriterConfig`. Application-level constructs use their own, lower-level constructs just use their encoding's default for now.